src/main/java/me/doubledutch/pikadb/PageCache.java

```diff
@@ -0,0 +1,85 @@
+package me.doubledutch.pikadb;
+
+import java.io.*;
+import java.nio.channels.*;
+import java.util.*;
+
+interface PageCache{
+	void Set(int id);
+}
+
+class NoopPageCache implements PageCache{
+	public void Set(int id) {
+		// Noop
+	}
+}
+
+class LRUPageCache implements PageCache{
+	private int size;
+	private Map<Integer, Page> pageMap;
+	private Deque<Integer> pageCache;
+
+	public LRUPageCache(Map<Integer, Page> map, int size){
+		this.size=size;
+		this.pageMap=map;
+		this.pageCache=new ArrayDeque<Integer>();
+	}
+
+	public void Set(int id){
+		if (!pageMap.containsKey(id)){
```

Comment:
Am I missunderstanding something - how will you ever get a new page in, when you exit if it's not already in?